aiorospy/src/aiorospy/action.py

```diff
@@ -28,3 +35,120 @@ def __init__(self, ns, action_spec, execute, loop=None):
     def _execute_cb(self):
         goal = self.accept_new_goal()
         future = asyncio.run_coroutine_threadsafe(self._execute(goal), self._loop)
+
+
+class AsyncGoalHandle:
+
+    def __init__(self, loop=None):
+        self.status = None
+        self.result = None
+
+        self._transition_q = janus.Queue(loop=loop)
+        self._feedback_q = janus.Queue(loop=loop)
+        self._old_statuses = set()
+
+        self._done_event = asyncio.Event()
+        self._status_event = asyncio.Event()
+
+        asyncio.create_task(self._process_transitions())
+
+    async def feedback(self):
+        while True:
+            terminal_status = asyncio.create_task(self._done_event.wait())
+            new_feedback = asyncio.create_task(self._feedback_q.async_q.get())
+            done, pending = await asyncio.wait(
+                {terminal_status, new_feedback},
+                return_when=asyncio.FIRST_COMPLETED)
+            if new_feedback in done:
+                yield new_feedback.result()
```

Comment:
async generators are one of the main reasons using Python 3.6 is important. Cuts down so much boilerplate.

Reply:
Yep, it's lovely